### PR TITLE
changes old intrinsic declaration to new declaration

### PR DESCRIPTION
**Pull request by aaishwarymishra**
This pr is for issue #132735 

It changes old `extern "intrinsic"` code block with new declaration.

There are other blocks that use old declaration but as the changes needed in single block is quite large I do them in parts

### CODE REVIEW

**Comment by BLANKatGITHUB** [comment]
Hello?

**Comment by RalfJung** [comment]
Looks good, thanks. :)

@bors r+

**Comment by jieyouxu** [comment]
[Failed in rollup](https://github.com/rust-lang/rust/pull/133378#issuecomment-2495466554), needs to rebless a UI test.

```
2024-11-23T12:41:36.0603872Z ---- [ui] tests/ui/intrinsics/reify-intrinsic.rs stdout ----
2024-11-23T12:41:36.0606692Z Saved the actual stderr to "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/intrinsics/reify-intrinsic/reify-intrinsic.stderr"
2024-11-23T12:41:36.0607757Z diff of stderr:
2024-11-23T12:41:36.0607954Z 
2024-11-23T12:41:36.0608235Z 22	   |         ^^^^^^^^^^^^^^^^^^^^^^^^^ cannot coerce intrinsics to function pointers
2024-11-23T12:41:36.0608765Z 23	   |
2024-11-23T12:41:36.0609346Z 24	   = note: expected fn pointer `unsafe extern "rust-intrinsic" fn(_) -> _`
2024-11-23T12:41:36.0610229Z -	                 found fn item `unsafe extern "rust-intrinsic" fn(_) -> _ {floorf32}`
2024-11-23T12:41:36.0610987Z +	                 found fn item `unsafe fn(_) -> _ {floorf32}`
2024-11-23T12:41:36.0611442Z 26	
2024-11-23T12:41:36.0611756Z 27	error: aborting due to 3 previous errors
```

@bors r-

**Comment by BLANKatGITHUB** [comment]
umm @jieyouxu  is there a problem?
Because I ran the tests/ui/intrinstics  in my mchine and all the tests passed

**Comment by jieyouxu** [comment]
Hi @BLANKatGITHUB, if you rebase against latest `master`, you should be able to observe the failure. I was not able to repro the failure locally with your branch immediately, but I am able to repro the failure when I rebased against latest `master`.

**Comment by BLANKatGITHUB** [comment]
Well I confirmed it and it indeed fails on latest master branch.
can you pls give a little insight on what is causing it and how can I resolve it

File: sql.txt
Comment (RalfJung) [inline sql.txt:1]: Something went wrong here, this file shouldn't be in the PR.

Please always double-check your commits before pushing them!

Comment (RalfJung) [reply sql.txt:1]: Now you have a commit adding the sql.txt file and another commit removing it. Please squash those into a single commit (or squash the entire PR into a single commit if you prefer).

Comment (BLANKatGITHUB) [reply sql.txt:1]: yeah I know sorry for that

**Comment by RalfJung** [comment]
@BLANKatGITHUB when you are done with your changes and the PR should be reviewed again, please post a message saying `@rustbot ready`

**Comment by BLANKatGITHUB** [comment]
well the changes are ready , the code block that contains the 
`extern "intrinsic"`
is 100s of lines and the changes that big becomes confusing so I will prefer making another pr for that but if you want I can add those changes to this pr depends on you.

**Comment by RalfJung** [comment]
No, changing all the cases in the test suite is better done separately. But generally reviewers won't look at PRs that are marked "waiting for author", and your PR is marked as such. So let's do:

@rustbot ready

**Comment by RalfJung** [comment]
Looks good, thanks. :)

@bors r+

**Pull request merged**